### PR TITLE
Dexsearch: Broadcast a message returned with parse: as well

### PR DESCRIPTION
**Pull request by bumbadadabum**

### CODE REVIEW

**Comment by bumbadadabum** [comment]
fuck this doesn't work what the hell going to figure that out first

**Comment by bumbadadabum** [comment]
Ok fixed

File: chat-plugins/dexsearch.js [hunk 1]
```diff
@@ -40,12 +40,13 @@ if (!process.send) {
 				cmd: 'dexsearch',
 				canAll: (!this.broadcasting || room.isPersonal),
 				message: (this.broadcasting ? "" : message),
-			}).then(message => {
-				if (message.reply) {
-					this.sendReplyBox(message.reply);
-				} else if (message.parse) {
-					this.parse(message.parse);
+			}).then(response => {
```
Comment (Zarel) [inline chat-plugins/dexsearch.js]: Let's move `canBroadcast` here

File: chat-plugins/dexsearch.js [hunk 2]
```diff
@@ -31,8 +31,11 @@ if (!process.send) {
 		ds: 'dexsearch',
 		dsearch: 'dexsearch',
 		dexsearch: function (target, room, user, connection, cmd, message) {
-			if (!this.canBroadcast()) return;
-
+			if (this.broadcasting && !user.can('broadcast', null, room)) {
```
Comment (Zarel) [inline chat-plugins/dexsearch.js]: `this.broadcasting` will always be `false`.

You're looking for `cmdToken`

Comment (Zarel) [inline chat-plugins/dexsearch.js]: We might be better off adding another argument to `canBroadcast` though. Or adding another possible case to the existing argument.

**Pull request merged**